modules/oracle/blockTick.go

```diff
@@ -0,0 +1,60 @@
+package oracle
+
+import (
+	"log"
+	"slices"
+	"vsc-node/lib/utils"
+	"vsc-node/modules/common"
+	"vsc-node/modules/db/vsc/elections"
+	stateEngine "vsc-node/modules/state-processing"
+)
+
+func (o *Oracle) blockTick(bh uint64, headHeight *uint64) {
+	if headHeight == nil {
+		return
+	}
+
+	slotInfo := stateEngine.CalculateSlotInfo(bh)
+	schedule := o.stateEngine.GetSchedule(slotInfo.StartHeight)
+
+	var witnessSlot *stateEngine.WitnessSlot
+	for _, slot := range schedule {
+		if slot.SlotHeight == slotInfo.StartHeight {
+			witnessSlot = &slot
+			break
+		}
+	}
+
+	// get elected members
+	result, err := o.electionDb.GetElectionByHeight(*headHeight)
+	if err != nil {
+		log.Println("[oracle] failed to get currently elected members.", err)
+		return
+	}
+
+	members := result.ElectionDataInfo.Members
+	memberAccounts := utils.Map(
+		members,
+		func(e elections.ElectionMember) string { return e.Account },
+	)
+
+	var (
+		username              = o.conf.Get().HiveUsername
+		avgPriceBroadcastTick = *headHeight%priceOracleBroadcastInterval == 0
+		chainRelayTick        = *headHeight%btcChainRelayInterval == 0
```

Comment:
Small nitpick. It looks like this is being treated as a boolean yet it is not clear as `isWitness` and `isBlockProducer`